unity/Assets/Scripts/IK_Robot_Arm_Controller.cs

```diff
@@ -342,151 +347,117 @@ public void OnTriggerEnter(Collider col)
+                setPosition(moveTransform, positionReset);
 
-            //if we hit a sim object
-            if(staticCollided.simObjPhysics && !staticCollided.gameObject)
-            {
-                debugMessage = "Arm collided with static sim object: '" + staticCollided.simObjPhysics.name + "' arm could not reach target position: '" + target + "'.";
-            }
+                //if we hit a sim object
+                if(staticCollided.simObjPhysics && !staticCollided.gameObject)
+                {
+                    debugMessage = "Arm collided with static sim object: '" + staticCollided.simObjPhysics.name + "' arm could not reach target position: '" + targetPosition + "'.";
+                }
 
-            //if we hit a structural object that isn't a sim object but still has static collision
-            if(!staticCollided.simObjPhysics && staticCollided.gameObject)
-            {
-                debugMessage = "Arm collided with static structure in scene: '" + staticCollided.gameObject.name + "' arm could not reach target position: '" + target + "'.";
-            }
+                //if we hit a structural object that isn't a sim object but still has static collision
+                if(!staticCollided.simObjPhysics && staticCollided.gameObject)
+                {
+                    debugMessage = "Arm collided with static structure in scene: '" + staticCollided.gameObject.name + "' arm could not reach target position: '" + targetPosition + "'.";
+                }
 
-            staticCollided.collided = false;
+                staticCollided.collided = false;
 
-            controller.actionFinished(false, debugMessage);
-            yield break;
+                actionSuccess = false;
+        } else if (armOvershot) {
+            Debug.Log("stopping arm height - target was overshot");
+            debugMessage =  "arm height has overshot the target position";
+            actionSuccess = false;
         }
-
-        //if the option to stop moving when the sphere touches any sim object is wanted
-
-        // Removed StopMotionOnContact until someone needs it, also should be renamed to StopMotionOn(Magnet|Hand)Contact
-        // if(magnetSphereComp.isColliding && StopMotionOnContact)
-        // {
-        //     string debugMessage = "Some object was hit by the arm's hand";
-        //     controller.actionFinished(false, debugMessage);
-        //     yield break;
-        // }
-
-        
-
-        controller.actionFinished(true);
+        controller.actionFinished(actionSuccess, debugMessage);
     }
 
-    public IEnumerator moveArmTarget(PhysicsRemoteFPSAgentController controller, Vector3 target, float unitsPerSecond,  GameObject arm, bool returnToStartPositionIfFailed = false, string whichSpace = "arm") {
+    // Restricts front hemisphere for arm movement
+    private bool validArmTargetPosition(Vector3 targetWorldPosition) {
```

Comment:
So this will prevent the target position from being valid if behind the "base" joint of the arm, but we also need to restrict it so that  points beyond the forward hemisphere are not valid. If the targetWorldPosition's z is beyond the radius of this sphere. This radius is 0.6335, which is the distance from the base joint to the wrist.